kratos/utilities/derivatives_extension.h

```diff
@@ -0,0 +1,166 @@
+                           "GetZeroDerivativesDofsVector method. Please "
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual void GetFirstDerivativesDofsVector(std::size_t NodeId,
+                                               std::vector<Dof<double>::Pointer>& rVector,
+                                               ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_THROW_ERROR(std::runtime_error,
+                           "Calling DerivativesExtension base class "
+                           "GetFirstDerivativesDofsVector method. Please "
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual void GetSecondDerivativesDofsVector(std::size_t NodeId,
+                                                std::vector<Dof<double>::Pointer>& rVector,
+                                                ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_THROW_ERROR(std::runtime_error,
+                           "Calling DerivativesExtension base class "
+                           "GetSecondDerivativesDofsVector method. Please "
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual void GetZeroDerivativesVariables(std::vector<VariableData const*>& rVariables,
+                                             ProcessInfo& rCurrentProcessInfo) const
+    {
+        KRATOS_THROW_ERROR(std::runtime_error,
+                           "Calling DerivativesExtension base class "
+                           "GetZeroDerivativesVariables method. Please "
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual void GetFirstDerivativesVariables(std::vector<VariableData const*>& rVariables,
+                                              ProcessInfo& rCurrentProcessInfo) const
+    {
+        KRATOS_THROW_ERROR(std::runtime_error,
+                           "Calling DerivativesExtension base class "
+                           "GetFirstDerivativesVariables method. Please "
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual void GetSecondDerivativesVariables(std::vector<VariableData const*>& rVariables,
+                                               ProcessInfo& rCurrentProcessInfo) const
+    {
+        KRATOS_THROW_ERROR(std::runtime_error,
+                           "Calling DerivativesExtension base class "
+                           "GetSecondDerivativesVariables method. Please "
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual std::ostream& Print(std::ostream& os) const
+    {
+        return os;
+    }
+
+private:
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const
+    {
+    }
+
+    void load(Serializer& rSerializer)
+    {
+    }
```

Comment:
I don't think this class should be needing/using the serializer, why did you add it?

Reply:
Anything that can be saved on a variable must be serializable, otherwise you would not be able to serialize nodes/elements.

Reply:
hm ok
It does not even compile if it does not have those methods right?